src/musicbrainz/tagfetcher.cpp

```diff
@@ -38,15 +38,15 @@ void TagFetcher::startFetch(
 
 void TagFetcher::abortAcoustIdTask() {
     if (m_pAcoustIdTask) {
-        disconnect(m_pAcoustIdTask.get());
+        disconnect(m_pAcoustIdTask);
         m_pAcoustIdTask->deleteBeforeFinished();
```

Comment:
The crash happens here. The operator-> returns the plain pointer here in the GUI thread. If the thread is suspended and the delete from the worker thread happens in between, than the pointer becomes dangling an the crash occurs.  
https://github.com/uklotzde/mixxx/pull/11 fixes the issue. 
I strongly recommend to remove all auto deletion code as done in the PR. 
If you need fire and forget nature in your other branches, inhered from the classes and add the desired nature.
It is kind of unpredictable and hard to review to have bot concepts mixed.

Reply:
The self-destruction code is now enclosed in VERFIY_AND_DEBUG_ASSERT to ensure that no unhandled signals slip through. It only serves as a fallback to prevent memory leaks.

I still don't understand how a crash could happen when using deleteLater(). This should ensure that objects remain alive until all signal handling has finished and control has been returned to the event loop.

Reply:
I hope I can explain: 

DeleteLater sends a signal to the object that calls a slot that deletes the object from the thread the object lives in. This ensures that all signals that have been issued before and might be still in the queue are executed before deletion, which would lead to a crash in the best case. 

Since Qt5 Qt uses it's QWeakPointer and QSharedPointer internally to maintain the object tree. This works with less locking than the Qt4 approach, that utilizes signals and slots. 
A QPointer is essential a wrapper around Qt's weak pointer. It becomes a nullptr at any time once the last reference to the object was removed. 

The code of question can be also written like this:   
```
    if (m_pAcoustIdTask) {
        disconnect(m_pAcoustIdTask);
        m_pAcoustIdTask->deleteBeforeFinished();
``` 
Can be also written inlined as: 
```
    if (m_pAcoustIdTask) {
        disconnect(m_pAcoustIdTask);
        mixxx::AcoustIdLookupTask pTask = m_pAcoustIdTask.data();
       QMetaObject::invokeMethod([] {pTask->slotAbort();  })
```
The crash happens if the object is deleted from the other thread just between the last two lines. m_pAcoustIdTask.data() returns the still valid pointer 0x120697a0 but it becomes  dangling before pTask->slotAbort() is called. 

The DeleteLater trick dos not help here, because the object is already gone.